### PR TITLE
jsampson's stuff from #16 plus some of mine

### PR DESCRIPTION
**Pull request by cklemkeberlin**
Reviewed and slightly modified changes as per pull request #16 plus some additional modifications

### CODE REVIEW

**Comment by jsampson** [comment]
Did you run into any actual problems with the boxing or unboxing in #16?

**Comment by cklemkeberlin** [comment]
The oldest JDK/JRE I am using on my development machine is 8, thus I haven't. But I remember Java 6 handling boxing/unboxing in a different, less flexible way than Java 7 and its successors, and that conforms with the info from the Stack Overflow article I am referring to in my commit message. Please note that while the put() method of List<T> expects T as parameter, its get() and containsKey() methods  still use Object as key type, which is for backwards compatibility reasons, I suppose. And it is my understanding that auto-boxing/unboxing only works for int<->Integer, long<->Long etc in Java 6, but not for int<->Object etc. Still, I have to admit that this is the situation in theory and that I haven't actually tested it. If you like, I can set up a legacy Java 6 environment on an older box and see what happens ?

**Comment by jsampson** [comment]
Got it, thanks. I would expect that building with `-source 1.6` should account for that, but it's worth testing. I actually already added a note to our 2.7 planning board that we should test against Java 6 before releasing. I'll go ahead and merge #16 now and then take a closer look at your further changes in the next few days, and we can test against Java 6 a little later.

File: pom.xml [hunk 1]
```diff
@@ -396,6 +396,30 @@
             <autoVersionSubmodules>true</autoVersionSubmodules>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.eclipse.m2e</groupId>
+          <artifactId>lifecycle-mapping</artifactId>
```
Comment (jsampson) [inline pom.xml]: Thanks for including the link in the commit message. As I understand it, this doesn't actually change how the build proceeds, but only adds some metadata for when Eclipse opens the project.

File: core/src/main/java/org/prevayler/implementation/PrevaylerDirectory.java [hunk 2]
```diff
@@ -160,7 +160,7 @@ public boolean accept(File pathname) {
 
     Arrays.sort(journals, new Comparator<File>() {
       public int compare(File f1, File f2) {
-        return new Long(journalVersion(f1)).compareTo(new Long(journalVersion(f2)));
+        return Long.valueOf(journalVersion(f1)).compareTo(Long.valueOf(journalVersion(f2)));
```
[the remark this answers is not in the document]
Comment (cklemkeberlin) [reply core/src/main/java/org/prevayler/implementation/PrevaylerDirectory.java]: Sorry for replying so late. I've been busy doing other stuff lately. Actually, I'm still quite new to Git (I've used Subversion for many years) and I first need to figure out how rebasing works.

[the remark this answers is not in the document]
Comment (cklemkeberlin) [reply core/src/main/java/org/prevayler/implementation/PrevaylerDirectory.java]: I'll have a look at this by the end of the week, ok ?

[the remark this answers is not in the document]
Comment (jsampson) [reply core/src/main/java/org/prevayler/implementation/PrevaylerDirectory.java]: Yeah, that sounds great. We're all super busy, so we're not in a hurry to finish the release.

If you're interested in figuring out Git rebasing, please feel free to take your time and use this as a learning experience. :) Don't hesitate to ask questions.

If not, just let me know and I can do the rebasing myself. I'll just need to know the email address that you want on the commits (which should be one associated with your GitHub account, either your actual email address or the special one from the "Keep my email address private" option).

(To be clear, the specific comment that you're replied to here, about `Long.compare`, is not something to change right now. The next one, about `Integer.parseInt`, is an actual suggestion.)

**Comment by jsampson** [comment]
I went ahead and rebased this branch, leaving out the Eclipse-related changes to pom.xml since I don't have an easy way to test them, but keeping all the code changes, and merged it into master: 107fdb38966d1df52728735ca8dc672ac4bdbe68

Thanks again!

**Pull request closed without merging**